sambaplayersdk/src/main/java/com/sambatech/player/plugins/TrackingLive.java

```diff
@@ -0,0 +1,313 @@
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Timer;
+import java.util.TimerTask;
+
+class TrackingLive implements Tracking {
+
+
+    private static final String EVENT_LOAD = "lo";
+    private static final String EVENT_PLAY = "pl";
+    private static final String EVENT_PAUSE = "pa";
+    private static final String EVENT_ONLINE = "on";
+    private static final String EVENT_COMPLETE = "co";
+
+    private static  final long TASK_EVENT_ON_TIME = 60000;
+
+    private static final String ORIGIN_SDK_ANDROID = "player.sambatech.sdk.android";
+
+    private SambaMediaConfig media;
+    private SttmLive sttmLive;
+
+    private Context context;
+
+    private SambaPlayerListener playerListener = new SambaPlayerListener() {
+
+        @Override
+        public void onLoad(SambaEvent event) {
+            super.onLoad(event);
+            sttmLive.trackLoadEvent();
+        }
+
+        @Override
+        public void onPlay(SambaEvent event) {
+            super.onPlay(event);
+
+            if (!sttmLive.isOnEventTaskRunning()) {
+                sttmLive.trackPlayAndONEvent();
+                sttmLive.startOnEventTask();
+            }
+
+        }
+
+        @Override
+        public void onPause(SambaEvent event) {
+            super.onPause(event);
+            sttmLive.cancelOnEventTask();
+            sttmLive.trackPauseEvent();
+        }
+
+        @Override
+        public void onError(SambaEvent event) {
+            super.onError(event);
+            sttmLive.cancelOnEventTask();
+        }
+    };
+
+
+    public void onLoad(@NonNull SambaPlayer player) {
+        Log.i("track", "load");
+
+        this.context = player.getContext();
+        this.media = (SambaMediaConfig) player.getMedia();
+
+        init();
+
+        PluginManager.getInstance().notifyPluginLoaded(this);
+    }
+
+    public void onInternalPlayerCreated(@NonNull SimpleExoPlayerView internalPlayer) {
+    }
+
+    public void onDestroy() {
+        Log.i("track", "destroy");
+        SambaEventBus.unsubscribe(playerListener);
+
+        if (sttmLive != null) {
+            sttmLive.destroy();
+            sttmLive = null;
+        }
+    }
+
+    private void init() {
+        if (media.sttmUrl != null && sttmLive == null) {
+            sttmLive = new SttmLive();
+        }
+
+        if (media.projectHash != null && media.id != null) {
+            SambaEventBus.subscribe(playerListener);
+        }
+    }
+
+    private Sttm2 getSttm2() throws Exception {
+
+        InputStream inputStream = null;
+        Scanner scanner = null;
+
+        try {
+
+            URL myURL = new URL(getSttm2RequestUrl());
+            HttpURLConnection conn = (HttpURLConnection) myURL.openConnection();
+            conn.setRequestProperty("Content-Type", "text/plain");
+
+            int responseCode = conn.getResponseCode();
+            if (responseCode >= 200 && responseCode < 299) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 200 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.sambatech.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 299 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.sambatech.com/coding_rules#rule_key=squid%3AS109)